### PR TITLE
add andTo and >+> to ZLayer

### PR DESCRIPTION
**Pull request by felher**
Here is my attempt to add the `>+>` operator to the ZLayer ops which allows for the linear chaining of topologically sorted layers.

Please let me know if we need tests for this. I'm happy to provide them but wasn't sure we need them, given that it's really only an alias for `++` and `>>>`, which are both covered by tests already.

Also, if I went overboard with the documentation, please let me know and I'll reduce it.

One last thing: If we want to settle on another name, I'm happy to make that change, too.

### CODE REVIEW

File: core/shared/src/main/scala/zio/ZLayer.scala [hunk 1]
```diff
@@ -2053,6 +2053,38 @@ object ZLayer {
...
+     *
+     * def all =
+     *   baker >+>       // provides: Baker
+     *   ingredients >+> // provides: Baker & Ingredients
+     *   oven >+>        // provides: Baker & Ingredients & Oven
+     *   dough >+>       // provides: Baker & Ingredients & Oven & Dough
+     *   cake            // provides: Baker & Ingredients & Oven & Dough & Cake
+     * }}}
+     */
+    def >+>[E1 >: E, RIn2 >: ROut, ROut1 >: ROut, ROut2 <: Has[_]](
```
Comment (adamgfraser) [inline core/shared/src/main/scala/zio/ZLayer.scala:2076]: These two methods should have the same signature.

File: core/shared/src/main/scala/zio/ZLayer.scala [hunk 2]
```diff
@@ -2053,6 +2053,38 @@ object ZLayer {
...
+      that: ZLayer[RIn2, E1, ROut2]
+    )(implicit tagged: Tagged[ROut2]): ZLayer[RIn, E1, ROut1 with ROut2] =
+      self ++ (self >>> that)
+
+    /**
+     * A named alias for `>+>`.
+     */
+    def andTo[E1 >: E, RIn2 >: ROut, ROut2 <: Has[_]](
```
Comment (adamgfraser) [inline core/shared/src/main/scala/zio/ZLayer.scala:2084]: Alphabetize with symbolic operators before alphanumeric methods.

File: core/shared/src/main/scala/zio/ZLayer.scala [hunk 3]
```diff
@@ -2053,6 +2053,38 @@ object ZLayer {
      */
     def update[A: Tagged](f: A => A)(implicit ev: ROut <:< Has[A]): ZLayer[RIn, E, ROut] =
       self >>> ZLayer.fromFunctionMany(_.update[A](f))
+
+    /**
+     * Returns a new layer that keeps the output of this layer and also adds the output of the other layer,
```
Comment (adamgfraser) [inline core/shared/src/main/scala/zio/ZLayer.scala:2058]: I would reduce this.

Comment (adamgfraser) [reply core/shared/src/main/scala/zio/ZLayer.scala:2058]: Maybe:
```scala
/**
   * Feeds the output services of this layer into the input of the specified
   * layer, resulting in a new layer with the inputs of this layer, and the
   * outputs of both this layer and the specified layer.
   */
```

File: core/shared/src/main/scala/zio/ZLayer.scala (hunk 2 above)
Comment (felher) [reply core/shared/src/main/scala/zio/ZLayer.scala]: Done :+1:

File: core/shared/src/main/scala/zio/ZLayer.scala (hunk 3 above)
Comment (felher) [reply core/shared/src/main/scala/zio/ZLayer.scala]: Done. Thanks for reviewing :)

Comment (mkotsbak) [reply core/shared/src/main/scala/zio/ZLayer.scala]: The cake example could be useful to include somwhere else, like at website in page https://zio.dev/docs/howto/howto_use_layers I don't think the documentation there is complete enough yet.

Comment (adamgfraser) [reply core/shared/src/main/scala/zio/ZLayer.scala]: Yes, I think we could use a section there discussing passed through versus hidden dependencies.

Comment (adamgfraser) [reply core/shared/src/main/scala/zio/ZLayer.scala]: I am working on now.

**Pull request merged**